openapi-adoc/src/main/java/io/micronaut/openapi/adoc/utils/CollectionUtils.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.openapi.adoc.utils;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * CollectionUtils.
+ *
+ * @since 5.2.0
+ */
+public final class CollectionUtils {
```

Comment:
is this needed there is already this type in `micronaut-core` module

Reply:
Yes, I know, but the idea is to not depend on this library.

By the way, I saw that it will still appear in the dependencies, because... it is added by the Build plugin. How to correctly remove this background dependency?

same problem in other modules: `openapi-annotations`, `openapi-generator`...